src/resolver/resolver.zig

```diff
@@ -1338,7 +1338,16 @@ pub const Resolver = struct {
                 }
             }
 
-            var source_dir_info = (r.dirInfoCached(source_dir) catch null) orelse return .{ .not_found = {} };
+            // If the source directory doesn't exist, walk up to find the first one that does, if any.
+            var source_dir_info = first_dir_info: {
+                var source_dir_to_use = source_dir;
+                while (true) {
+                    if (r.dirInfoCached(source_dir_to_use) catch return .{ .not_found = {} }) |dir|
+                        break :first_dir_info dir;
+                    source_dir_to_use = std.fs.path.dirname(source_dir) orelse
+                        return .{ .not_found = {} };
+                }
+            };
```

Comment:
my worry on this loop is that dirInfoCached does not seem to cache not found entries, which might be worth doing in some places

Reply:
It does cache not found entries. No need to worry.